File: check_braces.py
import re

def strip_content(text):
    # Remove comments
    text = re.sub(r'//.*', '', text)
    text = re.sub(r'/\*.*?\*/', '', text, flags=re.DOTALL)
    # Remove strings (handle escapes)
    text = re.sub(r'"(\\.|[^"\\])*"', '""', text)
    text = re.sub(r"'(\\.|[^'\\])*'", "''", text)
    text = re.sub(r"`(\\.|[^`\\])*`", "``", text, flags=re.DOTALL)
    # Remove regex literals (basic)
    text = re.sub(r'/(?!/| \*)(?:\\.|[^/\\])+/', '//', text)
    return text

with open('supabase/functions/uazapi-webhook/index.ts', 'r') as f:
    lines = f.readlines()

stack = []
for i, line in enumerate(lines):
    stripped = strip_content(line)
    for char in stripped:
        if char == '{':
            stack.append(('{', i+1))
        elif char == '}':
            if not stack:
                print(f"Extra }} at line {i+1}")
            else:
                stack.pop()
        elif char == '(':
            stack.append(('(', i+1))
        elif char == ')':
            if not stack:
                print(f"Extra ) at line {i+1}")
            elif stack[-1][0] == '(':
                stack.pop()
            else:
                print(f"Mismatched ) at line {i+1}, expected {stack[-1][0]} from line {stack[-1][1]}")
                stack.pop()

for char, line in stack:
    print(f"Unclosed {char} from line {line}")

File: check_braces.ts
import fs from 'fs';

const content = fs.readFileSync('supabase/functions/uazapi-webhook/index.ts', 'utf8');
const stack = [];
const lines = content.split('\n');

for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  for (let j = 0; j < line.length; j++) {
    const char = line[j];
    if (char === '{' || char === '(' || char === '[') {
      stack.push({ char, line: i + 1, col: j + 1 });
    } else if (char === '}' || char === ')' || char === ']') {
      if (stack.length === 0) {
        console.log(`Unexpected ${char} at line ${i + 1}, col ${j + 1}`);
      } else {
        const last = stack.pop();
        if ((char === '}' && last.char !== '{') ||
            (char === ')' && last.char !== '(') ||
            (char === ']' && last.char !== '[')) {
          console.log(`Mismatched ${char} at line ${i + 1}, col ${j + 1} (expected matching ${last.char} from line ${last.line})`);
        }
      }
    }
  }
}

if (stack.length > 0) {
  console.log('Unclosed brackets:');
  stack.forEach(s => console.log(`${s.char} at line ${s.line}, col ${s.col}`));
}
